src/services/admission/admission-router.test.ts

```diff
@@ -0,0 +1,79 @@
+
+const TESTER_USER = {
+    userId: TESTER.id,
+    status: DecisionStatus.ACCEPTED,
+    response: DecisionResponse.PENDING,
+    emailSent: false,
+    reviewer: "tester-reviewer",
+} satisfies DecisionInfo;
+
+const OTHER_USER = {
+    userId: "other-user",
+    status: DecisionStatus.REJECTED,
+    response: DecisionResponse.DECLINED,
+    emailSent: true,
+    reviewer: "other-reviewer",
+} satisfies DecisionInfo;
+
+beforeEach(async () => {
+    Models.initialize();
+    await Models.DecisionInfo.create(TESTER_USER);
+    await Models.DecisionInfo.create(OTHER_USER);
+});
+
+describe("GET /admission", () => {
+    it("gives forbidden error for user without elevated perms", async () => {
+        const responseUser = await getAsUser("/admission/").expect(StatusCode.ClientErrorForbidden);
+        expect(JSON.parse(responseUser.text)).toHaveProperty("error", "InvalidToken");
+    });
+    it("gives forbidden error for user without elevated perms - attendee", async () => {
+        const responseAttendee = await getAsAttendee("/admission/").expect(StatusCode.ClientErrorForbidden);
+        expect(JSON.parse(responseAttendee.text)).toHaveProperty("error", "InvalidToken");
+    });
+    it("should return a list of applicants without email sent", async () => {
+        const response = await getAsStaff("/admission/").expect(StatusCode.SuccessOK);
+        expect(JSON.parse(response.text)).toMatchObject(expect.arrayContaining([expect.objectContaining(TESTER_USER)]));
+    });
+});
+
+describe("PUT /admission", () => {
+    const updateData = [
+        {
+            userId: TESTER.id,
+            name: TESTER.name,
+            status: DecisionStatus.WAITLISTED,
+        },
+        {
+            userId: "other-user",
+            name: "other-name",
+            status: DecisionStatus.ACCEPTED,
+        },
+    ];
+    it("gives forbidden error for user without elevated perms (As User)", async () => {
+        const responseAttendee = await putAsAttendee("/admission/").send(updateData).expect(StatusCode.ClientErrorForbidden);
+        expect(JSON.parse(responseAttendee.text)).toHaveProperty("error", "InvalidToken");
+    });
+    it("gives forbidden error for user without elevated perms (As Attendee)", async () => {
+        const responseUser = await putAsUser("/admission/").send(updateData).expect(StatusCode.ClientErrorForbidden);
+        expect(JSON.parse(responseUser.text)).toHaveProperty("error", "InvalidToken");
+    });
+    it("should update application status of applicants", async () => {
+        const response = await putAsStaff("/admission/").send(updateData).expect(StatusCode.SuccessOK);
+        expect(JSON.parse(response.text)).toHaveProperty("message", "StatusSuccess");
+        const ops = updateData.map((entry) => {
+            return Models.DecisionInfo.findOne({ userId: entry.userId });
+        });
+        const retrievedEntries = await Promise.all(ops);
+        updateData.forEach((entry) => {
```

Comment:
Don't do forEach, do a single expect call here.

```ts
expect(retrivedEntries).toMatchObject(
    expect.arrayContaining(updateData)
)
```

Or, if you want partial equality:
```ts
expect(retrivedEntries).toMatchObject(
    expect.arrayContaining(updateData.map(
        (item) => expect.objectContaining(item)
    )
)
```